test/addons_test.go

```diff
@@ -1,103 +1,256 @@
 package test
 
 import (
+	"context"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
 	"testing"
 
 	"github.com/blang/semver"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/google/uuid"
+	"gopkg.in/yaml.v2"
+
+	volumetypes "github.com/docker/docker/api/types/volume"
+	docker "github.com/docker/docker/client"
+
+	"sigs.k8s.io/kind/pkg/apis/config/v1alpha3"
 
 	"github.com/mesosphere/kubeaddons/hack/temp"
 	"github.com/mesosphere/kubeaddons/pkg/api/v1beta1"
+	"github.com/mesosphere/kubeaddons/pkg/repositories/local"
 	"github.com/mesosphere/kubeaddons/pkg/test"
 	"github.com/mesosphere/kubeaddons/pkg/test/cluster/kind"
 )
 
-// TestAddons tests deployment of all addons in this repository
-func TestAddons(t *testing.T) {
-	cluster, err := kind.NewCluster(semver.MustParse("1.16.3"))
+const (
+	defaultKubernetesVersion = "1.16.4"
+	patchStorageClass        = `{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}`
+)
+
+var addonTestingGroups = make(map[string][]string)
+
+func init() {
+	b, err := ioutil.ReadFile("groups.yaml")
 	if err != nil {
-		t.Fatal(err)
+		panic(err)
```

Comment:
Why are we using panic instead of t.Fatal?

Reply:
The diff is bad, this line was not actually changed for this.

Reply:
Ah, one of _those_ PRs. :stuck_out_tongue_closed_eyes: